coralproject/talk
# [CORL-8] prevent repeat comments

- stores ID of user's most recent comment in redis for 6 hours 
- adds moderation step to check the body of that comment and compare it against the new comment
- throws an error if we're nudging, flags comment and witholds if not
- display "Duplicate comment" flag on comments marked as duplicates

**Question:** the expiration of the last id in redis... it makes sense in this context but I'm wondering if we might want this to be more general-purpose and have the ID _not_ expire and instead compare the createdAt in the moderation step? But then we're storing one entry indefinitely per user in redis and that could be a lot? 

How to test: 
1. log in as commenter on non-premod stream
2. post a comment
3. attempt to post the same comment
4. observe nudge, post comment anyways
5. observe message that comment will be held for moderation
6. log in as admin, comment should be marked "Duplicate comment"
